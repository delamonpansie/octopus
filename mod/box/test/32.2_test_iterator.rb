#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

class Env < RunEnv
  def config
    super + <<EOD
object_space[0].enabled = 1
object_space[0].index[0].type = "TREE"
object_space[0].index[0].unique = 1
object_space[0].index[0].key_field[0].fieldno = 0
object_space[0].index[0].key_field[0].type = "STR"

EOD
  end
end

Env.connect_eval do
  keys = %w{00 aa abc abcd abce def}
  keys.each do |i| insert_nolog [i] end

  lua 'user_proc.iterator', '0', '', keys.length.to_s # all keys
  lua 'user_proc.iterator', '0', 'ab', '3' # iterator from non-existing key must mathing prefix
end
