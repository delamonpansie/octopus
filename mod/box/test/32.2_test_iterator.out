# box.lua("user_proc.iterator", "0", "", "6")
[["00"], ["aa"], ["abc"], ["abcd"], ["abce"], ["def"]]

# box.lua("user_proc.iterator", "0", "ab", "3")
[["abc"], ["abcd"], ["abce"]]

